flask-api/flaskr/auth.py

```diff
@@ -40,4 +47,43 @@ def register():
     )
     db.commit()
     # return redirect(url_for('auth.login'))
-    return("good shit")
+    return "good shit"
+
+
+@bp.route('/login', methods=['POST'])
+def login():
+    try:
+        username = request.json['username']
+        password = request.json['password']
+    except:
+        abort(400, 'Bad Request')
+
+    db = get_db()
+
+    user = db.execute(
+        'SELECT * FROM user_cred WHERE username = ?', (username,)
+    ).fetchone()
+
+    if user is None:
+        abort(400, 'Incorrect username.')
+    elif not check_password_hash(user['password'], password):
+        abort(400, 'Incorrect password.')
+
+    # do JWT stuff here
+    access_token = create_access_token(identity=username)
+    resp = jsonify({'login': True})
+    set_access_cookies(resp, access_token)
+    return resp, 200
+
+
+@bp.route('/logout', methods=['POST', 'GET'])
+def logout():
+    resp = jsonify({'logout': True})
+    unset_jwt_cookies(resp)
+    return resp, 200
+
+
+@bp.route('/test', methods=['POST', 'GET'])
+@jwt_required

```

Comment:
@benedict-soh this is all u need else where for JWT validation. Just dont forget to import the package.